lib/rom/sql/attribute.rb

```diff
@@ -263,6 +292,17 @@ def __cmp__(op, other)
 
         Sequel::SQL::BooleanExpression.new(op, self, value)
       end
+
+      # Unwraps an optional type
+      #
+      # @api private
+      def unwrap_type(type)
+        if type.respond_to?(:left) && type.left == Types::Strict::Nil
```

Comment:
this can stay for now but please add `TODO: add Type#optional? method to dry-types and clean this up`